src/python/grpcio/grpc/_channel.py

```diff
@@ -1123,7 +1123,8 @@ def reset_postfork_child(self):
         self.managed_calls = 0
 
     def __del__(self):
-        if hasattr(self, 'channel') and self.channel:
+        if hasattr(self,
+                   'channel') and self.channel and cygrpc and cygrpc.StatusCode:
```

Comment:
Is there a chance that `self.channel.close` has been garbage collected? Maybe add `and hasattr(self.channel, "close")`? Same for `cygrpc.StatusCode` and `cygrpc.StatusCode.cancelled`

Reply:
If `self.channel` exists, then its subordinary objects should be alive. In the close function, there are many more usages of objects outside of this module. We can't prevent all corner cases, unless we suppress all `AttributeError` from this deallocation method.

Some parts of GC should still work even during interpreter shutdown, this PR is betting that.